packages/design-system/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,436 @@
+   */
+  escapeExitDisabled?: boolean;
+  /**
+   * Additional classes to be added to the header, which wraps the heading and
+   * close button.
+   */
+  headerClassName?: string;
+  /**
+   * The Dialog's heading, to be rendered in the header alongside the close button.
+   */
+  heading?: React.ReactNode;
+  /**
+   * The Dialog's size parameter.
+   */
+  size?: DialogSize;
+  /**
+   * @hide-prop [Deprecated] This prop has been renamed to `heading`.
+   */
+  title?: React.ReactNode;
+  /**
+   * Allow additional AriaModal props to be passed to Dialog
+   */
+  // eslint-disable-next-line camelcase
+  additional_props?: AriaModalProps;
+}
+
+export const Dialog = (props:DialogProps) :React.ReactNode => {
+  
+  let headingRef = useRef(null);
+  let eventHeadingText = '';
+
+  if (process.env.NODE_ENV !== 'production') {
+    if (props.title) {
+      console.warn(
+          `[Deprecated]: Please remove the 'title' prop in <Dialog>, use 'heading' instead. This prop has been renamed and will be removed in a future release.`
+          );
+    }
+    if (props.escapeExitDisabled) {
+      console.warn(
+          `[Deprecated]: Please remove the 'escapeExitDisabled' prop in <Dialog>, use 'escapeExits' instead. This prop has been renamed and will be removed in a future release.`
+          );
+    }
+    if (props.closeText) {
+      console.warn(
+          `[Deprecated]: Please remove the 'closeText' prop in <Dialog>, use 'closeButtonText' instead. This prop has been renamed and will be removed in a future release.`
+          );
+    }
+  }
+
+
+  useEffect(() => {
+    if (dialogSendsAnalytics() && props.analytics !== false) {
+      const heading = props.title || props.heading;
+
+      if (props.analyticsLabelOverride) {
+        eventHeadingText = props.analyticsLabelOverride;
+      } else if (typeof heading === 'string') {
+        eventHeadingText = heading.substring(0, MAX_LENGTH);
+      } else {
+        eventHeadingText =
+          headingRef.current?.textContent
+            ? headingRef.current.textContent.substring(0, MAX_LENGTH)
+            : '';
+      }
+
+      /* Send analytics event for dialog open */
+      sendLinkEvent({
+        event_name: 'modal_impression',
+        event_type: EVENT_CATEGORY.uiInteraction,
+        ga_eventAction: 'modal impression',
+        ga_eventCategory: EVENT_CATEGORY.uiComponents,
+        ga_eventLabel: eventHeadingText,
+        heading: eventHeadingText,
+      });
+    }
+
+    return () => {
+      if (dialogSendsAnalytics() && props.analytics !== false) {
+        /* Send analytics event for dialog close */
+        sendLinkEvent({
+          event_name: 'modal_closed',
+          event_type: EVENT_CATEGORY.uiInteraction,
+          ga_eventAction: 'closed modal',
+          ga_eventCategory: EVENT_CATEGORY.uiComponents,
+          ga_eventLabel: eventHeadingText,
+          heading: eventHeadingText,
+        });
+      }
+    }
+  },[])
+
+
+    const {
```

Comment:
Did this get ignored by Prettier? It looks like Prettier hasn't formatted this file

Reply:
Interesting, worked this time, i've been playing around with a git tool on the command line, must have gotten around it.. i've updated going through the normal linter/prettier